src/Integration.Vsix/SonarLintDaemon/CFamily.cs

```diff
@@ -0,0 +1,471 @@
+                        return "";
+                    case "true":
+                        return "/clr";
+                    case "Pure":
+                        return "/clr:pure";
+                    case "Safe":
+                        return "/clr:safe";
+                }
+            }
+
+            internal static string ConvertRuntimeLibrary(string value)
+            {
+                switch (value)
+                {
+                    default:
+                        throw new Exception("Unsupported RuntimeLibrary: " + value);
+                    case "MultiThreaded":
+                        return "/MT"; // defines macro "_MT"
+                    case "MultiThreadedDebug":
+                        return "/MTd"; // defines macros "_MT" and "_DEBUG"
+                    case "MultiThreadedDLL":
+                    case "MultiThreadedDll":
+                        return "/MD"; // defines macros "_MT" and "_DLL"
+                    case "MultiThreadedDebugDLL":
+                    case "MultiThreadedDebugDll":
+                        return "/MDd"; // defines macros "_MT", "_DLL" and "_DEBUG"
+                }
+            }
+
+            internal static string ConvertEnableEnhancedInstructionSet(string value)
+            {
+                switch (value)
+                {
+                    default:
+                        throw new Exception("Unsupported EnableEnhancedInstructionSet: " + value);
+                    case "NotSet":
+                        return "";
+                    case "AdvancedVectorExtensions":
+                        return "/arch:AVX";
+                    case "AdvancedVectorExtensions2":
+                        return "/arch:AVX2";
+                    case "StreamingSIMDExtensions":
+                        return "/arch:SSE";
+                    case "StreamingSIMDExtensions2":
+                        return "/arch:SSE2";
+                    case "NoExtensions":
+                        return "/arch:IA32";
+                }
+            }
+
+            internal static string ConvertExceptionHandling(string value)
+            {
+                switch (value)
+                {
+                    default:
+                        throw new Exception("Unsupported ExceptionHandling: " + value);
+                    case "false":
+                        return "";
+                    // all options below define macro "_CPPUNWIND":
+                    case "Async":
+                        return "/EHa";
+                    case "Sync":
+                        return "/EHsc";
+                    case "SyncCThrow":
+                        return "/EHs";
+                }
+            }
+
+            internal static string ConvertBasicRuntimeChecks(string value)
+            {
+                switch (value)
+                {
+                    default:
+                        throw new Exception("Unsupported BasicRuntimeChecks: " + value);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') 'System.Exception' should not be thrown by user code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=csharpsquid%3AS112)